todo_app/app.py

```diff
@@ -1,49 +1,127 @@
-from flask import Flask, render_template, request, redirect, url_for
+from flask import Flask, render_template, request, redirect, url_for, abort
+from flask_login import LoginManager, login_required
+import flask_login
+from flask_login.mixins import UserMixin
+from oauthlib.oauth2 import WebApplicationClient
+from oauthlib.oauth2.rfc6749.grant_types import client_credentials
 import Mongo_items as mongo
 import viewmodel as vm
+import os, requests, json
+from functools import wraps
 
 def create_app():
     app = Flask(__name__)
     app.config.from_object('flask_config.Config')
 
+    login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        client = WebApplicationClient(os.getenv("CLIENT_ID"))
+        authredirect = client.prepare_request_uri("https://github.com/login/oauth/authorize", state="saxasdfaa")
+        return redirect(authredirect)
+
+    @login_manager.user_loader
+    def load_user(user_id):
+        user = User(user_id)
+        return user
+
+    login_manager.init_app(app)
+
+    class User(UserMixin):
+        def __init__(self, userid):
+            self.user_id = userid
+            self.id = userid
+            if self.id == "69510597" :
+                self.role = 'reader'
+            else :
+                self.role = 'writer'
+
+    def checkrole(f):
+        @wraps(f)
+        def decorated_function(*args, **kwargs):
+
+            user = load_user(flask_login.current_user.id)
```

Comment:
This errors for an anonymous user - so your tests & application both error if the `LOGIN_DISABLED` flag is True (you may not see this if you've already signed in before switching the flag on, so try it in e.g. an incognito window). We can fix that a few ways;
* Call the get_id() function that does exist rather than the id field - e.g. `flask_login.current_user.get_id()`
* Add our own implementation of [AnonymousUser](https://flask-login.readthedocs.io/en/latest/#anonymous-users) that does have an id field
* Explicitly check the LOGIN_DISABLED flag to determine what to do here -  e.g. we might use a function to get the users role rather than loading them from the existing user, something like:
```python
    def get_user_role():
        if app.config['LOGIN_DISABLED']:
            return "writer"
        else:
            return flask_login.current_user.role
```

Reply:
Thanks!